peps/pep-0751.rst

```diff
@@ -0,0 +1,925 @@
+to install.
+
+The motivation for this approach to locking is for those who have controlled
+environments that they work with. For instance, if you have specific, controlled
+development and production environments then you can use per-file locking to
+make sure the **same** files are installed in either environment for everyone.
+
+This is similar to what ``pip freeze`` and pip-tools_
+support, but with more strictness of the exact files as well as incorporating
+support to specify the locked files for multiple environments in the same file.
+
+
+Package Locking
+===============
+
+*Package locking* lists the packages and their versions that *may* apply to any
+environment being installed for. The list of packages and their versions are
+evaluated individually and independently from any other packages and versions
+listed in the file. This allows installation to be linear -- read each package
+and version and make an isolated decision as to whether it should be installed.
+This avoids requiring the installer to perform a *resolution* (i.e.
+determine what to install based on what else is to be installed).
+
+The motivation of this approach comes from
+`PDM lock files <https://frostming.com/en/2024/pdm-lockfile/>`__. By listing the
+potential packages and versions that may be installed, what's installed is
+controlled in a way that's easy to reason about. This also allows for not
+specifying the exact environments that would be supported by the lock file so
+there's more flexibility for what environments are compatible with the lock
+file. This approach supports scenarios like open source projects that want to
+lock what people should use to build the documentation without knowing upfront
+what environments their contributors are working from.
+
+As already mentioned, this approach is supported by PDM_. Poetry_ has
+`shown some interest <https://discuss.python.org/t/lock-files-again-but-this-time-w-sdists/46593/83>`__.
+
+
+=============
+Specification
+=============
+
+---------
+File Name
+---------
+
+A lock file MUST be named ``pylock.toml`` or match the regular expression
+``f"pylock\.(.+)\.toml"`` if a name is desired or if multiple lock files exist.
+The use of the ``.toml`` file extension is to make syntax highlighting in
+editors easier and to reinforce the fact that the file format is meant to be
+human-readable. The prefix and suffix of a named file MUST be lowercase for easy
+detection and stripping off to find the name, e.g.::
+
+  if filename.startswith("pylock.") and filename.endswith(".toml"):
+      name = filename.removeprefix("pylock.").removesuffix(".toml")
+
+This PEP has no opinion as to the location of lock files (i.e. in the root or
+the subdirectory of a project).
+
+
+-----------
+File Format
+-----------
+
+The format of the file is TOML_.
+
+All keys listed below are required unless otherwise noted. If two keys are
+mutually exclusive to one another, then one of the keys is required while the
```

Comment:
Is there a way to know which is the required key and which is disallowed?

Reply:
It's documented for each key in the PEP where mutual exclusion applies. Are you after something else?